src/ast_pattern.mli

```diff
@@ -109,6 +109,18 @@ val __' : ('a, 'a Loc.t -> 'b, 'b) t
     In the latter case you should use the [pexp_loc] field of the captured
     expression instead. *)
 
+val drop : ('a, 'b, 'b) t
+(** Useful when we don't care about some part of OCaml AST. With [__] the actual
+    value will be passed to success continuation, with [drop] it will be ignore.
+    In normal pattern matching it is called wildcard pattern. *)
+
+val as__ : ('a, 'b, 'c) t -> ('a, 'a -> 'b, 'c) t
+(** As-pattern. Passed the current AST to success continuation.
+
+    Pitfall. The part of AST is being passed before checking that it fits the
+    pattern. It means that success continuation may fire earlier than we realize
+    that pattern argument of [as__] doesn't fit. *)
```

Comment:
Good observation! That shouldn't have a strong impact on the user though, should it? I mean, even if the success continuation set some global reference or similar, the program would stop right afterward. (Also, I'm wondering if the same happens in other combinators as well or if this is the only situation where that happens)

Reply:
> That shouldn't have a strong impact on the user though, should it? 

I had a bug about it, that took too much time to debug. 

> (Also, I'm wondering if the same happens in other combinators as well or if this is the only situation where that happens)

It looks like it is not being introduced by. I just was very disappointed about this pitfall, that's why this PR looks like it is.

Reply:
> I had a bug about it, that took too much time to debug.

That's annoying. Would it take long to explain the bug?

Reply:
I forgot to add `()`. I wrote like in `test.ml:62` but it was required to write like on line `test.ml:66`.

Reply:
Out of curiosity: What's your use case of `Ast_pattern`? I'm wondering because you seem to use the `parse` function yourself on the pattern as opposed to passing your pattern to `Extension.declare` or `Deriving.Args.arg` or similar.

Reply:
Ok, sounds interesting! And I can get an idea of why you're parsing your ast_pattern yourself.